Mathlib/Data/Matrix/Basic.lean

```diff
@@ -2017,6 +2017,13 @@ theorem transpose_map {f : α → β} {M : Matrix m n α} : Mᵀ.map f = (M.map
   rfl
 #align matrix.transpose_map Matrix.transpose_map
 
+theorem transpose_inj (A B : Matrix m n α) : Aᵀ = Bᵀ ↔ A = B :=
+  ⟨(fun h => ext fun i j => by simpa [transpose_apply] using ext_iff.2 h j i),
+    fun h => congr_arg transpose h ⟩
+
+theorem transpose_injective : Function.Injective (transpose : Matrix m n α → Matrix n m α) :=
```

Comment:
It will be easier to prove these in the other order using `eq_iff`

Reply:
Which kind of easy are we talking about. They are both one liners, and I according to my understanding lean compiles a theorem it's proof is irrelevant?